Source/WebCore/css/StyleProperties.cpp

```diff
@@ -705,6 +707,68 @@ String StyleProperties::fontVariantValue() const
     return result.toString();
 }
 
+String StyleProperties::fontSynthesisValue() const
+{
+    StringBuilder result;
+
+    auto getExplicitLonghandValue = [&](CSSPropertyID propertyID) -> CSSValue* {
+        auto foundPropertyIndex = findPropertyIndex(propertyID);
+        if (foundPropertyIndex == -1)
+            return nullptr;
+
+        auto property = propertyAt(foundPropertyIndex);
+        if (property.isImplicit())
+            return nullptr;
+
+        return property.value();
+    };
+
+    auto appendWithPrefixIfNeeded = [&](ASCIILiteral word) {
+        if (!result.isEmpty())
+            result.append(' ');
+        result.append(word);
+    };
+    
```

Comment:
Can you move this down closer to where it's used? e.g. before `if (valueID(weightValue) == CSSValueAuto)`